common/network-common/src/main/java/org/apache/spark/network/server/OneForOneStreamManager.java

```diff
@@ -124,6 +125,7 @@ public void connectionTerminated(Channel channel) {
       StreamState state = entry.getValue();
       if (state.associatedChannel == channel) {
         streams.remove(entry.getKey());
+        totalChunksBeingTransferred.addAndGet((-state.chunksBeingTransferred.get()));
```

Comment:
It is possible to get rid of `state.chunksBeingTransferred` totally?

Reply:
No we can't because without it, we do not know how many to remove from total when channel terminated.

Reply:
Maybe we can track transferring chunks by channel, e.g. `Map[Channel, Count]`? The number of channels should be less than `streams`.

And, if we can not get rid of `state.chunksBeingTransferred`, `connectionTerminated` could also be somewhat time consuming as it also traverse all `streams`.

Reply:
> And, if we can not get rid of state.chunksBeingTransferred, connectionTerminated could also be somewhat time consuming as it also traverse all streams.

IIUC, connectionTerminated won't be called frequently compared to chunksBeingTransferred which would less matter.
It only traverses the streams

Reply:
Oh, yeah. This a good point.